File: apps/ios/Barns/BarnsApp/Domain/Entities/ProductItem.swift
import Foundation

struct ProductItem: Equatable, Identifiable, Sendable {
    let id: String
    let categoryId: String
    let name: String
    let type: ProductItemType
    let installedOrPurchasedAt: Date?
    let locationLabel: String?
    let status: ProductItemStatus
    let careGuideIds: [String]
    let notes: String?
    let imageUrl: URL?
    let updatedAt: Date?
}

enum ProductItemType: String, Equatable, Sendable {
    case installed
    case purchased
}

enum ProductItemStatus: String, Equatable, Sendable {
    case active
    case archived
}

File: apps/android/app/src/main/java/com/barns/app/domain/model/ProductItem.kt
package com.barns.app.domain.model

import java.time.Instant

data class ProductItem(
    val id: String,
    val categoryId: String,
    val name: String,
    val type: ProductItemType,
    val installedOrPurchasedAt: Instant?,
    val locationLabel: String?,
    val status: ProductItemStatus,
    val careGuideIds: List<String>,
    val notes: String?,
    val imageUrl: String?,
    val updatedAt: Instant?,
)

enum class ProductItemType { INSTALLED, PURCHASED }

enum class ProductItemStatus { ACTIVE, ARCHIVED }
